netbox/utilities/filters.py

```diff
@@ -111,6 +117,152 @@ def __init__(self, *args, **kwargs):
+            if isinstance(existing_filter, (
+                MultiValueDateFilter,
+                MultiValueDateTimeFilter,
+                MultiValueNumberFilter,
+                MultiValueTimeFilter
+            )):
+                lookup_map = FILTER_NUMERIC_BASED_LOOKUP_MAP
+
+            elif isinstance(existing_filter, (
+                django_filters.ModelChoiceFilter,
+                django_filters.ModelMultipleChoiceFilter,
+                TreeNodeMultipleChoiceFilter,
+                TagFilter
+            )) or existing_filter.extra.get('choices'):
+                # These filter types support only negation
+                lookup_map = FILTER_NEGATION_LOOKUP_MAP
+
+            elif isinstance(existing_filter, (
+                django_filters.filters.CharFilter,
+                django_filters.MultipleChoiceFilter,
+                MultiValueCharFilter,
+                MultiValueMACAddressFilter
+            )):
+                lookup_map = FILTER_CHAR_BASED_LOOKUP_MAP
+
+            else:
+                # Do not augment any other filter types with more lookup expressions
+                continue
+
+            # Get properties of the existing filter for later use
+            field_name = existing_filter.field_name
+            field = get_model_field(cls._meta.model, field_name)
+
+            # Create new filters for each lookup expression in the map
+            for lookup_name, lookup_expr in lookup_map.items():
+                new_filter_name = '{}__{}'.format(existing_filter_name, lookup_name)
+                if existing_filter.lookup_expr == 'in':
+                    lookup_expr = 'in'  # 'in' lookups must remain to avoid unwanted slicing on certain querysets
+
+                try:
+                    if existing_filter_name in cls.declared_filters:
+                        # The filter field has been explicity defined on the filterset class so we must manually
+                        # create the new filter with the same type because there is no guarantee the defined type
+                        # is the same as the default type for the field
+                        resolve_field(field, lookup_expr)  # Will raise FieldLookupError if the lookup is invalid
+                        new_filter = type(existing_filter)(
+                            field_name=field_name,
+                            lookup_expr=lookup_expr,
+                            label=existing_filter.label,
+                            exclude=existing_filter.exclude,
+                            distinct=existing_filter.distinct,
+                            **existing_filter.extra
+                        )
+                    else:
+                        # The filter field is listed in Meta.fields so we can safely rely on default behaviour
+                        new_filter = cls.filter_for_field(field, field_name, lookup_expr)
+                except django_filters.exceptions.FieldLookupError:
```

Comment:
Wondering if we should move the `try`/`except` inside `if existing_filter_name in cls.declared_filters`. Seems like if `cls.filter_for_field()` fails on a default field it should make some noise.

Reply:
`cls.filter_for_field()` will also raise `FieldLookupError` if a field explicit does not support a lookup expression we are trying to apply to it. Internally it is calling `resolve_field()`. I have added a comment to make this clear.